### PR TITLE
Clean and bump all dependencies

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/CustomerBundle/composer.json [hunk 1]
```diff
@@ -36,21 +36,18 @@
...
-        "egulias/email-validator": "^3.1",
+        "doctrine/orm": "^2.18",
+        "egulias/email-validator": "^4.0",
         "sylius/customer": "^2.0",
         "sylius/resource-bundle": "^1.12",
         "symfony/framework-bundle": "^6.4.1 || ^7.1",
-        "webmozart/assert": "^1.9"
-    },
-    "conflict": {
-        "doctrine/orm": ">= 3.0.0"
```
Comment (Wojdylak) [inline src/Sylius/Bundle/CustomerBundle/composer.json:47]: Why did you remove this conflict?

**Review by Wojdylak — DISMISSED** [review]

File: composer.json [hunk 2]
```diff
@@ -31,50 +31,49 @@
...
+        "liip/imagine-bundle": "^2.13",
         "pagerfanta/pagerfanta": "^4.0",
-        "payum/offline": "^1.7.3",
-        "payum/payum-bundle": "^2.5",
+        "payum/offline": "^1.7.5",
+        "payum/payum-bundle": "^2.6",
+        "php-http/discovery": "^1.20",
         "php-http/guzzle7-adapter": "^1.0",
         "php-http/httplug": "^2.4",
         "php-http/message-factory": "^1.0",
```
Comment (mpysiak) [inline composer.json]: Can we maybe get rid of this package? 

Screenshot 2024-10-21 at 15 50 09

**Review by mpysiak — DISMISSED** [review]

File: src/Sylius/Bundle/CustomerBundle/composer.json (hunk 1 above)
Comment (GSadee) [reply src/Sylius/Bundle/CustomerBundle/composer.json:47]: Because in this bundle, we have in the main requirements section: "doctrine/orm": "^2.18", so this explicit requirement restricts the installation of a higher version than 2.x

File: composer.json (hunk 2 above)
Comment (GSadee) [reply composer.json]: It doesn't seem to be that easy because of Payum, so for now, I've just moved it to dev dependencies 🖖🏻 
```
request.CRITICAL: Uncaught PHP Exception Symfony\Component\Messenger\Exception\HandlerFailedException: "Handling "Sylius\Bundle\PayumBundle\PaymentRequest\Command\CapturePaymentRequest" failed: No php-http message factories found. Note that the php-http message factories are deprecated in favor of the PSR-17 message factories. To use the legacy Guzzle, Diactoros or Slim Framework factories of php-http, install php-http/message and php-http/message-factory and the chosen message implementation." at DoctrineTransactionMiddleware.php line 42
```

**Review by mpysiak — APPROVED** [review]
